certificate.go

```diff
@@ -93,24 +93,23 @@ func (c Certificate) Expires() time.Time {
 
 // GetFingerprints returns the list of certificate fingerprints, one of which
 // is computed with the digest algorithm used in the certificate signature.
-func (c Certificate) GetFingerprints() []DTLSFingerprint {
+func (c Certificate) GetFingerprints() ([]DTLSFingerprint, error) {
 	fingerprintAlgorithms := []dtls.HashAlgorithm{dtls.HashAlgorithmSHA256}
 	res := make([]DTLSFingerprint, len(fingerprintAlgorithms))
 
 	i := 0
 	for _, algo := range fingerprintAlgorithms {
 		value, err := dtls.Fingerprint(c.x509Cert, algo)
 		if err != nil {
-			fmt.Printf("Failed to create fingerprint: %v\n", err)
-			continue
+			return nil, fmt.Errorf("failed to create fingerprint: %v", err)
```

Comment:
Are you sure you want to do this change here? This seems to change the behaviour. If the change is correct maybe clarify it on the commit and/or PR for traceability?

Reply:
@maxhawkins 

I would rather fail fast on this one (I just can't think of a case where something like this would fail and I wouldn't want to know)

Also I have never seen this error message printed before (and haven't gotten user feedback around it) but I do think I would rather things fail then move past it.

Happy to split into its own commit though, if you both are good with this change.